sdk/monitor/monitor-query/src/generated/logquery/src/azureLogAnalyticsContext.ts

```diff
@@ -26,15 +26,19 @@ export class AzureLogAnalyticsContext extends coreHttp.ServiceClient {
       options = {};
     }
 
-    if (!options.userAgent) {
-      const defaultUserAgent = coreHttp.getDefaultUserAgentValue();
-      options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
-    }
+    const defaultUserAgent = `azsdk-js-${packageName.replace(
+      "@azure/",
+      ""
+    )}/${packageVersion} ${coreHttp.getDefaultUserAgentValue()}`;
 
-    super(undefined, options);
+    super(undefined, {
+      ...options,
+      userAgent: options.userAgent
```

Comment:
Assuming this had a user-agent defined in the convenience layer client constructor, this would double-up the user-agent string. I would just make sure the user-agent looks good and is not doubled